bondarenko/hw_unit_3_4/src/main/java/filters/AuthorizationFilter.java

```diff
@@ -0,0 +1,47 @@
+package filters;
+
+
+import javax.servlet.*;
+import javax.servlet.annotation.WebFilter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+@WebFilter(filterName = "AuthorizationFilter", urlPatterns = "/*")
+public class AuthorizationFilter implements Filter{
+
+    private Set<String> uris = new HashSet<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        uris.add("/login.jsp");
+        uris.add("/registration.jsp");
+    }
+
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+        HttpSession session = req.getSession(false);
+        if (session == null && !isNeededToPass(req, resp, filterChain)){
+            resp.sendError(HttpServletResponse.SC_FORBIDDEN);
+        } else {
+            filterChain.doFilter(servletRequest, servletResponse);
+        }
+    }
+
+    private boolean isNeededToPass(HttpServletRequest req, HttpServletResponse resp, FilterChain filterChain) throws IOException, ServletException {
+        return uris.contains(req.getRequestURI());
```

Comment:
why this method need response and filterChain? Also, you don't need full request, just get uri and pass the string